ui/src/app/dashboard/views/DiscoveriesList/DiscoveriesList.tsx

```diff
@@ -71,74 +84,80 @@ const generateRows = (
       className: classNames("p-table__row", {
         "is-active": isExpanded,
       }),
-      columns: [
-        {
-          content: (
-            <>
-              {name}
-              {discovery.is_external_dhcp ? (
-                <Tooltip
-                  message="This device is providing DHCP"
-                  className="u-nudge-right"
-                  position="top-center"
-                >
-                  <Icon name="information" />
-                </Tooltip>
-              ) : null}
-            </>
-          ),
-        },
-        {
-          content: (
-            <DoubleRow
-              primary={discovery.mac_address}
-              secondary={discovery.mac_organization || "Unknown"}
-            />
-          ),
-        },
-        {
-          content: discovery.ip,
-        },
-        {
-          content: discovery.observer_hostname,
-        },
-        {
-          content: <div className="u-truncate">{discovery.last_seen}</div>,
-        },
-        {
-          content: (
-            <ContextualMenu
-              data-test="row-menu"
-              hasToggleIcon={true}
-              links={[
-                {
-                  children: "Add discovery...",
-                  "data-test": "add-discovery-link",
-                  onClick: () => {
-                    setExpandedRow({
-                      id: discovery[DiscoveryMeta.PK],
-                      type: ExpandedType.ADD,
-                    });
-                  },
-                },
-                {
-                  children: "Delete discovery...",
-                  "data-test": "delete-discovery-link",
-                  onClick: () => {
-                    setExpandedRow({
-                      id: discovery[DiscoveryMeta.PK],
-                      type: ExpandedType.DELETE,
-                    });
-                  },
-                },
-              ]}
-              toggleClassName="row-menu-toggle u-no-margin--bottom"
-              toggleAppearance="base"
-            />
-          ),
-          className: "u-align--right",
-        },
-      ],
+      columns:
+        // Hide the columns when the add form is visible.
+        isExpanded && expandedRow?.type === ExpandedType.ADD
+          ? []
```

Comment:
I know this is how the legacy version works but it seems like a weird pattern to hide the column data and then show it again in a different place. We could also probably add the network details to the columns if we used double rows. And the name field being above the rest of the fields, for some reason? Just a bit weird.

Reply:
I actually started to rework it to be consistent with newer parts of the app and then saw that domains does that same name field at the top thing. I'll have another look, but might be good to get some design guidance on it.